src/main/scala/util/SBox.scala

```diff
@@ -0,0 +1,344 @@
+// See LICENSE.SiFive for license details.
+//
+//  This is translated from https://github.com/riscv/riscv-crypto
+//
+//    2020-01-29    Markku-Juhani O. Saarinen <mjos@pqshield.com>
+//    Copyright (c) 2020, PQShield Ltd. All rights reserved.
+//    Converted to Chisel in 2022
+
+/*
+    [BoPe12] Boyar J., Peralta R. "A Small Depth-16 Circuit for the AES
+    S-Box." Proc.SEC 2012. IFIP AICT 376. Springer, pp. 287-298 (2012)
+    DOI: https://doi.org/10.1007/978-3-642-30436-1_24
+    Preprint: https://eprint.iacr.org/2011/332.pdf
+
+    [Ny93] Nyberg K., "Differentially Uniform Mappings for Cryptography",
+    Proc. EUROCRYPT '93, LNCS 765, Springer, pp. 55-64 (1993)
+    DOI: https://doi.org/10.1007/3-540-48285-7_6
+*/
+
+package freechips.rocketchip.util
+
+import chisel3._
+
+// top (inner) linear layer for AES
+object SBoxAESEncIn {
```

Comment:
I really like the SBox impl here! wondering if is possible to upstream this to `chisel.stdlib`.

Reply:
Will do so